### PR TITLE
Interceptors batteries included

### PR DESCRIPTION
**Pull request by werdes72**
**Description**

Changes proposed in this pull request:

- allow interceptors to handle default domain & tls certificate for non local cluster (not provisioned via gardener)

### CODE REVIEW

File: parallel-install/pkg/deployment/overrideinterceptors_test.go [hunk 1]
```diff
@@ -224,71 +281,299 @@ func Test_DomainNameOverrideInterceptor(t *testing.T) {
...
+		ob := OverridesBuilder{}
+
+		ob.AddInterceptor([]string{"global.tlsCrt", "global.tlsKey"}, interceptor)
+
+		// when
+		overrides, err := ob.Build()
+
+		// then
 		require.NoError(t, err)
+		require.Empty(t, overrides.Map()) //No override injected
```
Comment (mjakobczyk) [inline parallel-install/pkg/deployment/overrideinterceptors_test.go:400]: The assertion is pretty straight-forward. Do we need the comment here? Or maye move it to the line starting with `// then..`?

**Review by mjakobczyk — CHANGES_REQUESTED** [review]

File: parallel-install/pkg/deployment/overrideinterceptors_test.go [hunk 2]
```diff
@@ -175,47 +183,96 @@ func Test_GlobalOverridesInterception(t *testing.T) {
...
+}
+
 func Test_DomainNameOverrideInterceptor(t *testing.T) {
 	ob := OverridesBuilder{}
 	log := logger.NewLogger(true)
 
-	domainData := make(map[string]string)
-	domainData["domain"] = "gardener.domain"
+	//domainData := make(map[string]string)
+	//domainData["domain"] = "gardener.domain"
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply parallel-install/pkg/deployment/overrideinterceptors_test.go:220]: Removed

File: parallel-install/pkg/deployment/overrideinterceptors_test.go [hunk 3]
```diff
@@ -175,47 +183,96 @@ func Test_GlobalOverridesInterception(t *testing.T) {
...
+		kubeClient := fake.NewSimpleClientset()
+		ob.AddInterceptor([]string{"global.domainName"}, mockNewDomainNameOverrideInterceptor(kubeClient, log, false))
+
+		// when
+		overrides, err := ob.Build()
+
+		// then
+		require.NoError(t, err)
+		require.NotEmpty(t, overrides)
+		require.Contains(t, overrides.String(), defaultRemoteKymaDomain)
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply parallel-install/pkg/deployment/overrideinterceptors_test.go:255]: Done.

File: parallel-install/pkg/deployment/overrideinterceptors_test.go [hunk 4]
```diff
@@ -175,47 +183,96 @@ func Test_GlobalOverridesInterception(t *testing.T) {
...
+	require.Equal(t, expected, overrides.Map())
+}
+
 func Test_DomainNameOverrideInterceptor(t *testing.T) {
 	ob := OverridesBuilder{}
 	log := logger.NewLogger(true)
 
-	domainData := make(map[string]string)
-	domainData["domain"] = "gardener.domain"
+	//domainData := make(map[string]string)
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply parallel-install/pkg/deployment/overrideinterceptors_test.go:219]: Removed

File: parallel-install/pkg/deployment/overrideinterceptors_test.go [hunk 5]
```diff
@@ -175,47 +183,96 @@ func Test_GlobalOverridesInterception(t *testing.T) {
...
+		// given
+		kubeClient := fake.NewSimpleClientset()
+		ob.AddInterceptor([]string{"global.domainName"}, mockNewDomainNameOverrideInterceptor(kubeClient, log, false))
+
+		// when
+		overrides, err := ob.Build()
+
+		// then
+		require.NoError(t, err)
+		require.NotEmpty(t, overrides)
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply parallel-install/pkg/deployment/overrideinterceptors_test.go:254]: We would then have to compare maps for equality and prepare a test map for every test case. I don't think it's necessary.

[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply parallel-install/pkg/deployment/overrideinterceptors_test.go:254]: But we can improve the assertion a bit without going that far. I'll add assertions on key count trying to ensure no additional entries appear in the map.

[the remark this answers is not in the document]
Comment (mjakobczyk) [reply parallel-install/pkg/deployment/overrideinterceptors_test.go:254]: Great, this is what I would see here!

File: parallel-install/pkg/deployment/overrideinterceptors_test.go [hunk 6]
```diff
@@ -224,71 +281,299 @@ func Test_DomainNameOverrideInterceptor(t *testing.T) {
...
+		require.Equal(t, getOverride(overrides.Map(), "global.tlsCrt"), defaultRemoteTLSCrtEnc)
+		require.Equal(t, getOverride(overrides.Map(), "global.tlsKey"), defaultRemoteTLSKeyEnc)
+	})
+
+	t.Run("test default cert is not set for a gardener cluster", func(t *testing.T) {
+		kubeClient := fake.NewSimpleClientset(gardenerCM)
+
+		// given
+		interceptor := NewCertificateOverrideInterceptor("global.tlsCrt", "global.tlsKey", kubeClient)
+		interceptor.isLocalCluster = isLocalClusterFunc(true) //Try to trick the code into local domain. Should be handled properly
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply parallel-install/pkg/deployment/overrideinterceptors_test.go:389]: Moved comment to test case description

File: parallel-install/pkg/deployment/overrideinterceptors_test.go (hunk 1 above)
Comment (Tomasz-Smelcerz-SAP) [reply parallel-install/pkg/deployment/overrideinterceptors_test.go:400]: Removed

**Review by mjakobczyk — APPROVED** [review]
LGTM! 👍

File: parallel-install/pkg/deployment/core.go [hunk 7]
```diff
@@ -38,19 +40,17 @@ type core struct {
...
-	overrides, err := ob.Build()
+func newCore(cfg *config.Config, overrides Overrides, kubeClient kubernetes.Interface, processUpdates func(ProcessUpdate)) (*core, error) {
+	isK3d, err := isK3dCluster(kubeClient)
 	if err != nil {
 		return nil, err
 	}
 
+	if isK3d {
+		cfg.Log.Infof("Running in K3d cluster: removing incompatible components '%s'", strings.Join(incompatibleLocalComponents, "', '"))
+		removeFromComponentList(cfg.ComponentList, incompatibleLocalComponents)
```
Comment (Tomasz-Smelcerz-SAP) [inline parallel-install/pkg/deployment/core.go:51]: Dead code, please remove.

**Comment by Tomasz-Smelcerz-SAP** [comment]
Please test:
- [ ] if kyma.example.com cert is accepted by the browser
- [ ] what is necessary to run fast-integration-tests (perhaps adding to /etc/hosts)

**Review by clebs — DISMISSED** [review]

File: parallel-install/pkg/deployment/core.go (hunk 7 above)
Comment (werdes72) [reply parallel-install/pkg/deployment/core.go]: Done.

**Review by Tomasz-Smelcerz-SAP — APPROVED** [review]

File: parallel-install/pkg/deployment/core.go [hunk 8]
```diff
@@ -132,31 +118,46 @@ func (i *core) processUpdateComponent(phase InstallationPhase, comp components.K
...
 	//hide certificate data
-	o.AddInterceptor([]string{"global.domainName", "global.ingress.domainName"}, NewDomainNameOverrideInterceptor(kubeClient, log))
-	o.AddInterceptor([]string{"global.tlsCrt", "global.tlsKey"}, NewCertificateOverrideInterceptor("global.tlsCrt", "global.tlsKey"))
+	ob.AddInterceptor([]string{"global.domainName", "global.ingress.domainName"}, NewDomainNameOverrideInterceptor(kubeClient, log))
+	ob.AddInterceptor([]string{"global.tlsCrt", "global.tlsKey"}, NewCertificateOverrideInterceptor("global.tlsCrt", "global.tlsKey", kubeClient))
 	// make sure we don't install legacy CRDs
-	o.AddInterceptor([]string{"global.installCRDs"}, NewInstallLegacyCRDsInterceptor())
+	ob.AddInterceptor([]string{"global.installCRDs"}, NewInstallLegacyCRDsInterceptor())
+
+	ob.AddInterceptor([]string{"global.installCRDs"}, NewInstallLegacyCRDsInterceptor())
```
Comment (Tomasz-Smelcerz-SAP) [inline parallel-install/pkg/deployment/core.go:158]: It's an error to register it twice

**Review by Tomasz-Smelcerz-SAP — CHANGES_REQUESTED** [review]

**Review by Tomasz-Smelcerz-SAP — APPROVED** [review]

**Comment by Tomasz-Smelcerz-SAP** [comment]
I installed it using custom-built cli on GKE, overrides look OK.

**Pull request merged**